source/extensions/transport_sockets/tls/context_impl.h

```diff
@@ -101,6 +101,14 @@ class ContextImpl : public virtual Envoy::Ssl::Context {
 
   std::vector<Ssl::PrivateKeyMethodProviderSharedPtr> getPrivateKeyMethodProviders();
 
+  // Called by verifyCallback to do the actual cert chain verification.
+  int doVerifyCertChain(X509_STORE_CTX* store_ctx, Ssl::SslExtendedSocketInfo* ssl_extended_info,
```

Comment:
My suggestion would be to refactor the cert verification code in ContextImpl to allow for the QUIC code to pass in a cert chain (as a vector of buffers of bytes, not X509 structs) and a hostname so that none of the x509 code needs to be in the ProofVerifier impl. (This might also mean adding an additional function to get error details.)

Reply:
I agree; it would be nice if all the boringssl `X509` code was contained here. But if it makes it messier, I'm ok with leaving it how it is.